ext/opentelemetry-exporter-cloud-trace/src/opentelemetry/tools/resource_detector.py

```diff
@@ -0,0 +1,44 @@
+import requests
+
+from opentelemetry.sdk.resources import Resource, ResourceDetector
+
+_GCP_METADATA_URL = (
+    "http://metadata.google.internal/computeMetadata/v1?recursive=true"
```

Comment:
Note: A trailing `/` after `v1` is needed to inspect the variables recursively. This is enforced in some environments and will return a HTTP `301` response if missing.

Reply:
This is a good repo to learn about how to use the Metadata server in practice: https://github.com/googleapis/gcp-metadata